Superlu master (post 6.0.0) breaks PETSc

Either #99 or #98 broke configuration in PETSc, see, e.g., https://gitlab.com/petsc/petsc/-/jobs/4430894576.

@xiaoyeli @prj- I moved this to an issues of it own. Let us continue here with the discussion.

I am not sure whether this was really caused by #98 or #99, as Doxygen seems to have a problem.

From the configure.log:

`CMake Error at /nfs/gce/globalscratch/svcpetsc/petsc-hash-pkgs/7820d1/share/cmake-3.26/Modules/FindDoxygen.cmake:734 (message):`

and 

`CMake Warning at /nfs/gce/globalscratch/svcpetsc/petsc-hash-pkgs/7820d1/share/cmake-3.26/Modules/FindDoxygen.cmake:492 (message):
  Unable to determine doxygen version: 1
Call Stack (most recent call first):
  /nfs/gce/globalscratch/svcpetsc/petsc-hash-pkgs/7820d1/share/cmake-3.26/Modules/FindDoxygen.cmake:655 (_Doxygen_find_doxygen)
  DOC/CMakeLists.txt:3 (find_package)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /usr/bin/doxygen)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by /usr/bin/doxygen)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /lib/x86_64-linux-gnu/libclang-14.so.13)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /lib/x86_64-linux-gnu/libclang-cpp.so.14)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /lib/x86_64-linux-gnu/libclang-cpp.so.14)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by /lib/x86_64-linux-gnu/libclang-cpp.so.14)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `CXXABI_1.3.13' not found (required by /lib/x86_64-linux-gnu/libLLVM-14.so.1)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /lib/x86_64-linux-gnu/libLLVM-14.so.1)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /lib/x86_64-linux-gnu/libLLVM-14.so.1)
/usr/bin/doxygen: /nfs/gce/software/custom/linux-ubuntu22.04-x86_64/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/gcc-8.5.0-zpsjiod/lib64/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by /lib/x86_64-linux-gnu/libLLVM-14.so.1)`

Looks like an issue with Doxygen in the runner.

@prj- Can you please do the following to help narrowing down the problem:
1. Can you pass to SuperLU's CMake call this option: `-DCMAKE_DISABLE_FIND_PACKAGE_Doxygen=TRUE`. This prevents CMake from running Doxygen.
2. Can you Git bisect the SuperLU commit triggering your problem?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 # Required version
-cmake_minimum_required(VERSION 3.9)
+cmake_minimum_required(VERSION 3.12)
 
 # Project Version
 project(SuperLU C)

--- a/DOC/CMakeLists.txt
+++ b/DOC/CMakeLists.txt
@@ -15,10 +15,10 @@ if(enable_doc)
 endif()
 
 doxygen_add_docs(doc
-                 "${_DEPENDENCY_ALL}"
                  "${CMAKE_CURRENT_SOURCE_DIR}/mainpage.txt"
                  "${CMAKE_CURRENT_SOURCE_DIR}/modules.txt"
                  "${PROJECT_SOURCE_DIR}/EXAMPLE"
                  "${PROJECT_SOURCE_DIR}/SRC"
                  "${PROJECT_SOURCE_DIR}/TESTING"
+                 "${_DEPENDENCY_ALL}"
                  COMMENT "Generate HTML documentation with Doxygen")

--- a/DOC/CMakeLists.txt
+++ b/DOC/CMakeLists.txt
@@ -14,11 +14,15 @@ if(enable_doc)
   set(_DEPENDENCY_ALL "ALL")
 endif()
 
-doxygen_add_docs(doc
-                 "${CMAKE_CURRENT_SOURCE_DIR}/mainpage.txt"
-                 "${CMAKE_CURRENT_SOURCE_DIR}/modules.txt"
-                 "${PROJECT_SOURCE_DIR}/EXAMPLE"
-                 "${PROJECT_SOURCE_DIR}/SRC"
-                 "${PROJECT_SOURCE_DIR}/TESTING"
-                 "${_DEPENDENCY_ALL}"
-                 COMMENT "Generate HTML documentation with Doxygen")
+if(DOXYGEN_FOUND)
+  doxygen_add_docs(doc
+                  "${CMAKE_CURRENT_SOURCE_DIR}/mainpage.txt"
+                  "${CMAKE_CURRENT_SOURCE_DIR}/modules.txt"
+                  "${PROJECT_SOURCE_DIR}/EXAMPLE"
+                  "${PROJECT_SOURCE_DIR}/SRC"
+                  "${PROJECT_SOURCE_DIR}/TESTING"
+                  "${_DEPENDENCY_ALL}"
+                  COMMENT "Generate HTML documentation with Doxygen")
+elseif(enable_doc)
+  message(WARNING "Disabling building documentation as Doxygen is missing.")
+endif()

